### PR TITLE
Add discussion document describing alternate git workflow proposal

### LINKED ISSUE #199: Git workflow scaling
As a Mineral UI developer, I would like an improved git workflow, so that I have more control over the release process.

*NOTE: There is likely to be a lot of bike-shedding here.  The goal is to make improvements while NOT overcomplicating the process.*

### PR DESCRIPTION
**Pull request by brentertz**
### Description

Add document describing outcome of a discussion regarding a proposed new git workflow.

### Motivation and context

As a Mineral UI developer, I would like an improved git workflow, so that I have more control over the release process.

closes #199 

### How to test

n/a

### Types of changes

- Other (provide details below)

docs

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support) - [n/a]
* [x] Automated tests written and passing - [n/a]
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered - [n/a]
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered - [n/a]
* [x] Documentation created or updated

### How does this PR make you feel?

Deep thoughts by Octocat

### CODE REVIEW

**Comment by coldpour** [comment]
amazing gif

File: docs/git-workflow.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Git workflow
+
+As a Mineral UI developer, I would like an improved git workflow, so that I have more control over the release process.
```
Comment (coldpour) [inline docs/git-workflow.md]: can you say more about what control you're missing in the current release process?

**Comment by brentertz** [comment]
This PR was a proposal based on several team member concerns that our current git workflow did not easily allow for certain capabilities/scenarios.  This alternate git workflow was suggested as a potential solution to address these concerns and solicit feedback from team members.  

Following discussion, it was determined that the process change does not add enough value to deviate from our current git workflow.  All things can be achieved in or current model, albeit a bit differently.

This document will be updated and saved in the docs/discussions for posterity.

💡  Ideally, a similar document should be created outlining how to perform all of the same tasks using our current git workflow.

**Comment by brentertz** [comment]
The document in this PR has been updated to reflect the outcome of our team discussion and has been moved to the `docs/discussions` section of the project.

**Comment by brentertz** [comment]
A [separate story](https://github.com/mineral-ui/mineral-ui/issues/205) has been added to document common tasks using our current git workflow.

**Pull request merged**